proxy/proxy_test.go

```diff
@@ -1031,6 +1031,78 @@ var _ = Describe("Proxy", func() {
 				})
 			})
 		})
+
+		Context("when attempting to establish TLS handshake with a TCP application", func() {
```

Comment:
Can we frame this test (description at least, maybe the test implementation as well?) so it is less specific to a TCP application. Changing the context description to something about how the TLS handshake is not reciprocated by the upstream (and a comment that we are using a TCP server that will not continue the TLS handshake) may be enough, just to make it clear this is not a TCP app only issue

Reply:
+1